### ISSUE TITLE
Enhancement: [no-require-imports] Add option `allowJson`

### ISSUE THREAD

**Issue by remcohaszing** [typescript-eslint/typescript-eslint#5314]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-require-imports/

### Description

If `module` in `tsconfig.json` is set to `Node16`, TypeScript 4.7 will convert `import … = require('…')` statements to require calls created using `_createRequire`.

As long as import assertions are not allowed in Node.js, I believe this is a nice syntax to read `package.json` files.

I propose to add an option named `allowJson` to `no-require-imports` that allows require imports of JSON files.

### Fail

```typescript
import lodash = require('lodash')
import lib = require('./lib')
```

### Pass

```typescript
These are currently also errored on


import lodashPkg = require('lodash/package.json')
import pkg = require('./package.json')
```

### Additional Info

In:

```ts
import pkg = require('./package.json')

console.log(pkg.version)
```

Out:

```js
import { createRequire as _createRequire } from "module";
const __require = _createRequire(import.meta.url);
const pkg = __require("./package.json");
console.log(pkg.version);
```

[TypeScript playground](https://www.typescriptlang.org/play?target=99&module=100#code/JYWwDg9gTgLgBGA1gczgXjlApgRwK7DYAUA5AHQD0YAhgMaLXJZkBWAzhAHYkCUAUH1pcOAG2YiIyIkmRkAblihtgXHkA)

**Comment by bradzacher** [comment]
Given that node14 will EOL in april next year - is it worth adding an option to support the old way of doing things when people will be moving to the new way relatively soon?

Looking into it - it's supported in node 16 behind a flag, but node16 will EOL a few months after node 14. It's fully supported in node18, which will EOL in 3 years.

I'm not entirely convinced that an option is the best of ideas given the above and given how infrequent a usecase it is to directly import JSON.
